added join discord button

mohammedrehan275 opened the pull request:
join discord button is added in the sidebar of the application
Resolve #28

mohammedrehan275 commented:
I tried adding custom colors but that doesn't seem to be working. Is there anything that I have to do before adding custom colors in tailwind CSS. I tried adding the discord background color as `bg-cornflower-blue-500` , but that isn't working.

YashKamboj commented:
@mohammedrehan275 Hey bro have you solved the issue , i am happy to help you out .

rishipurwar1 commented:
> @mohammedrehan275 Hey bro have you solved the issue , i am happy to help you out .

How did you add the custom color? After adding custom colors, you need to rerun the react app(npm start)
I would suggest you stick with the default color palette if it's not very necessary to import the custom color, try to find the matching color in the default color palette first.
Here is the link to the default color palette: https://tailwindcss.com/docs/customizing-colors
You'll also find information related to how to add custom colors.

mohammedrehan275 commented:
> 
> How did you add the custom color? After adding custom colors, you need to rerun the react app(npm start)
> I would suggest you stick with the default color palette if it's not very necessary to import the custom color, try to find the matching color in the default color palette first.
> Here is the link to the default color palette: https://tailwindcss.com/docs/customizing-colors
> You'll also find information related to how to add custom colors.
> 
> If you need any help, let me know.
> please join the discord community.

I have added the colors from the color palette itself and created PR for the same. You can review it. VS Code extensions (Prettier) might have styled the page a bit. 
The button is working and you can review it

rishipurwar1 commented:
> yes, here are the screenshots

Can you please place a button at the bottom of the sidebar and add a margin around the button? that would look better

mohammedrehan275 commented:
Will make the changes

rishipurwar1 commented:
> Will make the changes

And please update your with the main branch before pushing your code to avoid merge conflict.

mohammedrehan275 commented:
> > Will make the changes
> 
> And please update your code with the main branch before pushing your code to avoid merge conflict.

Yes, I'll make sure to update the code

mohammedrehan275 commented:
- [x] placed the button at the bottom of the sidebar
- [x] changed join discord community to join discord
- [x] added `target="_blank" rel="noopener noreferrer"`

screenshot below for reference
![discord button](https://user-images.githubusercontent.com/65021509/134465560-cfdc46c6-2c30-4ac2-97ff-2689eda378db.JPG)

mohammedrehan275 commented:
this pull request is created in the development branch
If you want me to create PR in main branch let me know.

rishipurwar1 commented:
t

> this pull request is created in the development branch
> If you want me to create PR in main branch let me know.

You did a great job.
There is some file conflict.
I would suggest you rebase your branch with the main branch.
step-1 git fetch origin
step-2 git rebase origin/main
and then check your changes working properly.

mohammedrehan275 commented:
I guess the conflicts should be removed now

rishipurwar1 commented on src/components/layouts/SideBar.js:
```diff
@@ -68,10 +68,25 @@ const Sidebar = () => {
+              >
+                Join Discord
+              </a>
+            </button>
```
I would like to suggest one thing instead of putting an anchor tag in the button, convert an anchor to a button and put an icon inside the anchor tag.
Remove the button tag and copy the tailwind classes from the button and paste it to an anchor tag.
Does it make sense to you?

mohammedrehan275 replied:
What are the advantages of converting anchor to a button?
Just curious to know.

rishipurwar1 replied:
Not a very big advantage, we don't need to use one extra tag.

mohammedrehan275 replied:
I have made changes and the button is now converted into a anchor tag

rishipurwar1 replied:
Button looks good on desktop but not on mobile and tablet.
In tablet, it should be square and in mobile, it should take up the whole width of the mobile menu. Just fix these two issues.
<img width="543" alt="Screenshot 2021-09-24 at 7 54 39 PM" src="https://user-images.githubusercontent.com/57035408/134691000-d3176219-8255-4b3d-8829-2b198a8452ca.png">
<img width="83" alt="Screenshot 2021-09-24 at 7 56 13 PM" src="https://user-images.githubusercontent.com/57035408/134691092-3f25843b-bc5b-4443-b8e7-6407b3914f1f.png">

mohammedrehan275 replied:
Will fix it
